worldedit-core/src/main/java/com/fastasyncworldedit/core/queue/IBatchProcessor.java

```diff
@@ -35,61 +35,99 @@ default boolean processGet(int chunkX, int chunkZ) {
+        int maxLayer = (maxY + 1) >> 4;
+        if (keepInsideRange) {
+            for (int layer = set.getMinSectionPosition(); layer <= minLayer; layer++) {
+                if (set.hasSection(layer)) {
+                    if (layer == minLayer) {
+                        char[] arr = set.loadIfPresent(layer);
+                        if (arr != null) {
+                            int index = (minY & 15) << 8;
+                            for (int i = 0; i < index; i++) {
+                                arr[i] = 0;
+                            }
+                        } else {
+                            arr = new char[4096];
                         }
+                        set.setBlocks(layer, arr);
                     } else {
-                        arr = new char[4096];
+                        set.setBlocks(layer, null);
                     }
-                    set.setBlocks(layer, arr);
-                } else {
-                    set.setBlocks(layer, null);
                 }
             }
-        }
-        int maxLayer = (maxY + 1) >> 4;
-        for (int layer = maxLayer; layer < set.getMaxSectionPosition(); layer++) {
-            if (set.hasSection(layer)) {
-                if (layer == minLayer) {
-                    char[] arr = set.loadIfPresent(layer);
-                    if (arr != null) {
-                        int index = ((maxY + 1) & 15) << 8;
-                        for (int i = index; i < arr.length; i++) {
-                            arr[i] = 0;
+            for (int layer = maxLayer; layer < set.getMaxSectionPosition(); layer++) {
+                if (set.hasSection(layer)) {
+                    if (layer == minLayer) {
+                        char[] arr = set.loadIfPresent(layer);
+                        if (arr != null) {
+                            int index = ((maxY + 1) & 15) << 8;
+                            for (int i = index; i < arr.length; i++) {
+                                arr[i] = 0;
+                            }
+                        } else {
+                            arr = new char[4096];
                         }
+                        set.setBlocks(layer, arr);
                     } else {
-                        arr = new char[4096];
+                        set.setBlocks(layer, null);
+                    }
+                }
+            }
+            try {
+                int layer = (minY - 15) >> 4;
+                while (layer < (maxY + 15) >> 4) {
+                    if (set.hasSection(layer)) {
+                        return true;
                     }
-                    set.setBlocks(layer, arr);
-                } else {
-                    set.setBlocks(layer, null);
+                    layer++;
                 }
+            } catch (ArrayIndexOutOfBoundsException exception) {
+                getLogger(IBatchProcessor.class).error("minY = {} , layer = {}", minY, ((minY - 15) >> 4), exception);
```

Comment:
Create a logger instance from the LogManagerCompat, otherwise we get the platform logger.